bin/namespace-reporter.rb

```diff
@@ -182,27 +191,54 @@ def memory_value(str)
   end
 end
 
-############################################################
+def text_output(ns)
+  puts
+  puts "Namespace: #{ns[:name]}"
+  puts
+  puts "  Request limit:\tCPU: #{ns[:max_requests][:cpu]},\tMemory: #{ns[:max_requests][:memory]}"
+  puts "  Requested:\t\tCPU: #{ns[:resources_requested][:cpu]},\tMemory: #{ns[:resources_requested][:memory]}"
+  puts
+  puts "  Num. containers:\t#{ns[:container_count]}"
+  puts "  Req. per-container:\tCPU: #{ns[:default_request][:cpu]},\tMemory: #{ns[:default_request][:memory]}"
+  puts
+  puts "  Resources in-use:\tCPU: #{ns[:resources_used][:cpu]},\tMemory: #{ns[:resources_used][:memory]}"
+  puts
+  puts "CPU values are in millicores (m). Memory values are in mebibytes (Mi)."
+  puts
+end
 
-name = ARGV.shift
+def parse_options
+  options = { format: TEXT_OUTPUT }
+
+  OptionParser.new do |opts|
+    opts.on("-n", "--namespace NAMESPACE", "Namespace name pattern (required)") do |ns|
+      options[:namespace] = ns
+    end
+
+    opts.on("-o", "--output FORMAT", "Output format (#{JSON_OUTPUT} | #{TEXT_OUTPUT})") do |fmt|
```

Comment:
@digitalronin hi, this is great. was going to suggest `optparse` before I saw this. Am trying it out atm. one thing you could add to enforce the format options and default to text is;
```
    opts.on("-o", "--output [FORMAT]", [JSON_OUTPUT, TEXT_OUTPUT], "Output format (#{JSON_OUTPUT} | #{TEXT_OUTPUT})") do |fmt|
      raise "Unknown output format #{fmt}" unless [JSON_OUTPUT, TEXT_OUTPUT].include?(fmt)
      options[:format] = fmt || 'text'
    end
```

Reply:
Thanks for that, @jsugarman I've updated the script with your change.